src/adapter/src/notice.rs

```diff
@@ -273,6 +277,14 @@ impl fmt::Display for AdapterNotice {
                     object_name.quoted()
                 )
             }
+            AdapterNotice::KeyNotEnforced { key, name } => {
+                write!(
+                    f,
+                    "cannot verify key ({}) for sink {}; sunk updates may not be unique by key",
```

Comment:
This is a good opportunity to use the DETAIL and HINT fields! What do you think about something like this?

```
WARNING:  upsert envelope key not validated to be unique
DETAIL: Materialize could not validate that the specified upsert envelope key (...) is a unique key of the underlying relation (...). The sink might produce multiple updates for the same key at the same time. This can confuse downstream consumers of the sink.
HINT: Learn more: https://materialize.com/s/error-upsert-envelope-key-not-validated
```

We can leave the HINT out for now, but once we have docs about how to resolve this situation, it'd be really great to link to them in the error message.

Reply:
> What do you think about something like this?

Seems better than a semicolon! And DETAIL definitely leaves a lot more room for a helpful explanation.

> We can leave the HINT out for now, but once we have docs about how to resolve this situation, it'd be really great to link to them in the error message.

I'm a big fan of linking to documentation in error messages, so this sounds great. (I thought about adding docs as part of this PR, but I'll wait until we reach a bit more consensus on the linked issue.)

Reply:
Implemented the `DETAILS` as suggested!